Kinvey/Kinvey/Realtime.swift

```diff
@@ -26,4 +28,254 @@ protocol RealtimeRouter {
     
     func unsubscribe(channel: String, context: AnyHashable)
     
+    func publish(
+        channel: String,
+        message: Any,
+        completionHandler: ((Result<Void, Swift.Error>) -> Void)?
+    )
+    
+}
+
+public class LiveStream<Type: BaseMappable> {
+    
+    private let name: String
+    private let client: Client
+    
+    private var substreamChannelNameMap = [String : String]()
+    
+    fileprivate let uuid = UUID()
+    
+    public init(name: String, client: Client = sharedClient) {
+        self.name = name
+        self.client = client
+    }
+    
+    private var userPromise: Promise<User> {
+        return Promise<User> { fulfill, reject in
+            if let user = client.activeUser {
+                fulfill(user)
+            } else {
+                reject(Error.invalidOperation(description: "Active User not found"))
+            }
+        }
+    }
+    
+    private var realtimeRouterPromise: Promise<(User, RealtimeRouter)> {
+        return userPromise.then { activeUser in
+            return Promise<(User, RealtimeRouter)> { fulfill, reject in
+                if let realtimeRouter = activeUser.realtimeRouter {
+                    fulfill(activeUser, realtimeRouter)
+                } else {
+                    reject(Error.invalidOperation(description: "Active User not register for realtime"))
+                }
+            }
+        }
+    }
+    
+    @discardableResult
+    public func grantStreamAccess(userId: String, acl: LiveStreamAcl, completionHandler: ((Result<Void, Swift.Error>) -> Void)? = nil) -> Request {
+        let request = client.networkRequestFactory.buildLiveStreamGrantAccess(streamName: name, userId: userId, acl: acl)
+        Promise<Void> { fulfill, reject in
+            request.execute() { (data, response, error) in
+                if let response = response, response.isOK, let _ = data {
+                    fulfill()
+                } else {
+                    reject(buildError(data, response, error, self.client))
+                }
+            }
+        }.then {
+            completionHandler?(.success())
+        }.catch { error in
+            completionHandler?(.failure(error))
+        }
+        return request
+    }
+    
+    public func send(userId: String, message: Type, completionHandler: ((Result<Void, Swift.Error>) -> Void)? = nil) {
+        realtimeRouterPromise.then { activeUser, realtimeRouter in
+            return Promise<(RealtimeRouter, String)> { fulfill, reject in
+                if let channelName = self.substreamChannelNameMap[userId] {
+                    fulfill(realtimeRouter, channelName)
+                } else {
+                    let request = self.client.networkRequestFactory.buildLiveStreamPublish(streamName: self.name, userId: userId)
```

Comment:
@heyzooi Is a request to publish being made with every call to `send` or `post`?